src/app/core/admin/user-details/user-details.component.ts

```diff
@@ -0,0 +1,70 @@
+import { Component, Inject, OnInit } from '@angular/core';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
+import { User } from '../../user/user';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { EntityMapperService } from '../../entity/entity-mapper.service';
+import { MatSnackBar } from '@angular/material/snack-bar';
+
+@Component({
+  selector: 'app-user-details',
+  templateUrl: './user-details.component.html',
+  styleUrls: ['./user-details.component.scss'],
+})
+export class UserDetailsComponent implements OnInit {
+  creating: boolean = true;
+  private readonly defaultPasswordValue: string = 'password';
+  private readonly defaultUsernameValue: string = 'username';
+
+  userForm: FormGroup = this.fb.group({
+    username: [{value: this.defaultUsernameValue, disabled: !this.creating}, Validators.required],
+    password: [this.defaultPasswordValue, Validators.required],
+    admin: [false],
+  });
+
+  constructor(
+    private dialogRef: MatDialogRef<UserDetailsComponent>,
+    @Inject(MAT_DIALOG_DATA) public user: User,
+    private fb: FormBuilder,
+    private entityMapper: EntityMapperService,
+    private snackBar: MatSnackBar,
+  ) { }
+
+  ngOnInit() {
+    this.creating = this.user === null;
+    if (! this.creating) {
+      this.userForm.patchValue({username: this.user.name});
+      // Username can't be edited after creation
+      this.userForm.get('username').disable();
+      this.userForm.patchValue({admin: this.user.isAdmin()});
+    }
+  }
+
+  saveUser() {
+    const {username, password, admin} = this.userForm.getRawValue();
+    if (this.creating) {
+      this.user = new User(username);
+    } else if (username !== this.user.name) {
+      // Can't be reached because changing username is not allowed
+      this.entityMapper.remove<User>(this.user);
```

Comment:
careful. this may have side effects if we every reference the user entity id (because now it will be different). To be honest, I am not sure why we disallow username changes but I think it's probably because the username directly makes part of the _id and we wanted to keep the id immutable.

Maybe we should disallow username changes in the form also?